AutoCollection/HttpRequests.ts

```diff
@@ -89,7 +91,13 @@ class AutoCollectHttpRequests {
                 // Note: Check for if correlation is enabled happens within this method.
                 // If not enabled, function will directly call the callback.
                 CorrelationContextManager.runWithContext(correlationContext, () => {
-                    if (this._isEnabled) {
+                    let shouldCollect: boolean = !(<any>request)[AutoCollectHttpRequests.alreadyAutoCollectedFlag];
```

Comment:
We should do this check before `runWithContext` to avoid introducing unused contexts and thus losing association with dependencies.